src/mmds/src/ns.rs

```diff
@@ -530,4 +530,64 @@ mod tests {
         let expected = Ipv4Addr::from(DEFAULT_IPV4_ADDR);
         assert_eq!(actual, expected);
     }
+
+    #[test]
+    fn test_break_speculative_check_detour_arp() {
+        let mut buf = [0u8; 2000];
+        let ip = Ipv4Addr::from(DEFAULT_IPV4_ADDR);
+        let other_ip = Ipv4Addr::new(5, 6, 7, 8);
+        let mac = MacAddr::from_bytes_unchecked(&[0; 6]);
+        let mut ns = MmdsNetworkStack::new(
+            mac,
+            ip,
+            DEFAULT_TCP_PORT,
+            NonZeroUsize::new(DEFAULT_MAX_CONNECTIONS).unwrap(),
+            NonZeroUsize::new(DEFAULT_MAX_PENDING_RESETS).unwrap(),
+        );
+
+        {
+            let mut eth =
+                EthernetFrame::write_incomplete(buf.as_mut(), mac, mac, ETHERTYPE_ARP).unwrap();
+            let mut arp = EthIPv4ArpFrame::from_bytes_unchecked(eth.inner_mut().payload_mut());
+            arp.set_tpa(other_ip);
+            let len = ns.write_arp_request(buf.as_mut(), false);
+
+            let mut buf2 = buf;
+            let mut eth2 =
```

Comment:
You should be able to use [shadowing](https://doc.rust-lang.org/rust-by-example/variable_bindings/scope.html) here, as `let mut eth = ...`.